packages/web-components/src/tabs/tabs.ts

```diff
@@ -0,0 +1,185 @@
+  }
+
+  /**
+   * getSelectedTabScale - gets the scale of the tab
+   */
+  private getTabScale(tab: Tab): number {
+    if (this.orientation === TabsOrientation.horizontal) {
+      return this.previousActiveTabData.width / tab.getBoundingClientRect().width;
+    } else return this.previousActiveTabData.height / tab.getBoundingClientRect().height;
+  }
+
+  /**
+   * clearAnimationProperties
+   * resets animation values to defaults. removes the animated css class
+   */
+  private clearAnimationProperties(tab: Tab) {
+    this.previousActiveTabData = { id: '', x: 0, y: 0, height: 0, width: 0 };
+    this.activeTabOffset = 0;
+    this.activeTabScale = 1;
+    tab.classList.remove('animated');
+  }
+
+  /**
+   * animationLoop
+   * runs through all the operations required for setting the tab active indicator to its starting location, ending location, and applying the animated css class to the tab.
+   * @param tab
+   */
+  private animationLoop(tab: Tab) {
+    // If selected, set the animation properties.
+    // get the offset for active indicator
+    // the offset at the start of the animation will be at the location of the previously selected tab
+    this.calculateAnimationProperties(tab);
+    // set the active indicator offset and scale of the active indicator to css vars
+    this.setTabScaleCSSVar();
+    this.setTabOffsetCSSVar();
+
+    // clear properties for next animation
+    this.clearAnimationProperties(tab);
+    this.previousActiveTabData = this.activeTabData;
+
+    // add the animate css class if the calculated offset is 0 and scale is 1
+    if (this.activeTabOffset === 0 && this.activeTabScale === 1) {
+      this.calculateAnimationProperties(tab);
+      tab.classList.add('animated');
+      // update the css vars
+      this.setTabScaleCSSVar();
+      this.setTabOffsetCSSVar();
+      // now when the css class is added the active indicator ::after element will animate
+      // from the previous tab location to its starting location
+    }
+  }
+
+  /**
+   * setTabData
+   * sets the data from the active tab onto the class. used for making all the animation calculations for the active tab indicator.
+   */
+  private setTabData(): void {
+    if (this.tabs && this.tabs.length > 0) {
+      const tabs = this.tabs as Tab[];
+      const activeTab = this.activetab || tabs[0];
+      const activeRect = activeTab?.getBoundingClientRect();
+      const parentRect = this.getBoundingClientRect();
+
+      this.activeTabData = {
+        id: activeTab.id,
+        x: activeRect.x - parentRect.x,
+        y: activeRect.y - parentRect.y,
+        height: activeRect.height,
+        width: activeRect.width,
+      } as TabData;
+
+      if (!this.previousActiveTabData?.id) {
+        this.previousActiveTabData = this.activeTabData;
+      }
+    }
+  }
+
+  private setTabOffsetCSSVar() {
+    document.documentElement.style.setProperty(TabTokenNames.tabIndicatorOffset, `${this.activeTabOffset}px`);
```

Comment:
What is `document.documentElement` in this case? Is it the root document? Is it the parent shadow DOM if nested? I'm a bit concerned on reaching outside the component to set this.

Reply:
I can update this to target the :host on this.styles. I was having trouble figuring out how to type the key for the css variable so I could use the enumerated object with the css variable names:

<img width="695" alt="image" src="https://user-images.githubusercontent.com/90428651/228334803-69730dbd-0a72-4eb0-b94b-4c75df0f616c.png">
<img width="920" alt="image" src="https://user-images.githubusercontent.com/90428651/228334903-5830e228-087f-47ef-a482-8ec1bec3f1b6.png">

Reply:
@eljefe223 do you know?